src/common/libflux/fripp.c

```diff
@@ -0,0 +1,456 @@
+/************************************************************\
+ * Copyright 2021 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <stdbool.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <syslog.h>
+#include <time.h>
+#include <unistd.h>
+
+#include "attr.h"
+#include "flog.h"
+#include "fripp.h"
+#include "reactor.h"
+
+#include "src/common/libczmqcontainers/czmq_containers.h"
+#include "src/common/libutil/iterators.h"
+
+#define FRIPP_MAX_PACKET_LEN 1440
+#define INTERNAL_BUFFSIZE 128
```

Comment:
Shouldn't the buffer be larger than the max packet size?

Reply:
The buffer is dynamically sized. It is initialized to the `INTERNAL_BUFFSIZE`, but that size is also used for the formatting of metrics in `fripp_packet_appendf` and for the `prefix` buffer.

Reply:
Ah OK, sorry I was confused there.

Reply:
Maybe it could benefit from a better name or a comment?